server/routers/matchTeam_router.js

```diff
@@ -37,11 +36,11 @@ exports.getRouter = function () {
 function getDefaultMatchesForTeam (teamNumber, stages) {
   const matches = []
 
-  for (const stage of stages) {
-    for (let i = 1; i <= stages[stage].matchAmount; i++) {
+  for (const index in stages) {
```

Comment:
It's better to use the `of` syntax, the problem was that when using the `of` syntax you get the value in the `stage` parameter so you should switch any instance of `stages[stage]` with `stage`.